include/crocoddyl/core/actions/unicycle.hxx

```diff
@@ -0,0 +1,82 @@
+///////////////////////////////////////////////////////////////////////////////
+// BSD 3-Clause License
+//
+// Copyright (C) 2018-2020, LAAS-CNRS, University of Edinburgh
+// Copyright note valid unless otherwise stated in individual files.
+// All rights reserved.
+///////////////////////////////////////////////////////////////////////////////
+
+#include "crocoddyl/core/utils/exception.hpp"
+
+namespace crocoddyl {
+template <typename Scalar>
+ActionModelUnicycleTpl<Scalar>::ActionModelUnicycleTpl()
+    : ActionModelAbstractTpl<Scalar>(boost::make_shared<StateVectorTpl<Scalar> >(3), 2, 5), dt_(0.1) {
+  cost_weights_ << 10., 1.;
+}
+template <typename Scalar>
+ActionModelUnicycleTpl<Scalar>::~ActionModelUnicycleTpl() {}
+template <typename Scalar>
+void ActionModelUnicycleTpl<Scalar>::calc(const boost::shared_ptr<ActionDataAbstractTpl<Scalar> >& data,
+                                          const Eigen::Ref<const typename MathBase::VectorXs>& x,
+                                          const Eigen::Ref<const typename MathBase::VectorXs>& u) {
+  if (static_cast<std::size_t>(x.size()) != state_->get_nx()) {
+    throw_pretty("Invalid argument: "
+                 << "x has wrong dimension (it should be " + std::to_string(state_->get_nx()) + ")");
+  }
+  if (static_cast<std::size_t>(u.size()) != nu_) {
+    throw_pretty("Invalid argument: "
+                 << "u has wrong dimension (it should be " + std::to_string(nu_) + ")");
+  }
+
+  ActionDataUnicycleTpl<Scalar>* d = static_cast<ActionDataUnicycleTpl<Scalar>*>(data.get());
+  const Scalar& c = std::cos(x[2]);
```

Comment:
It won't work for code generation. You should remove the std.

Reply:
Thanks.